autogen/agentchat/contrib/gpt_assistant_agent.py

```diff
@@ -143,68 +147,108 @@ def _invoke_assistant(
             # pass the latest system message as instructions
             instructions=self.system_message,
         )
-
-        run_response_messages = self._get_run_response(assistant_thread, run)
-        assert len(run_response_messages) > 0, "No response from the assistant."
-
-        response = {
-            "role": run_response_messages[-1]["role"],
-            "content": "",
-        }
-        for message in run_response_messages:
-            # just logging or do something with the intermediate messages?
-            # if current response is not empty and there is more, append new lines
-            if len(response["content"]) > 0:
-                response["content"] += "\n\n"
-            response["content"] += message["content"]
-
+        self.cancellation_requested = False
+        response = self._get_run_response(assistant_thread, run)
         self._unread_index[sender] = len(self._oai_messages[sender]) + 1
-        return True, response
+        if response["content"]:
+            return True, response
+        else:
+            return False, "No response from the assistant."
 
-    def _get_run_response(self, thread, run):
+    def _process_messages(self, assistant_thread, run):
+        """
+        Processes and provides a response based on the run status.
+        Args:
+            assistant_thread: The thread object for the assistant.
+            run: The run object initiated with the OpenAI assistant.
+        """
+        if run.status == "failed":
+            logger.error(f'Run: {run.id} Thread: {assistant_thread.id}: failed...')
+            if run.last_error:
+                response = {
+                    "role": "assistant",
+                    "content": str(run.last_error),
+                }
+            else:
+                response = {
+                    "role": "assistant",
+                    "content": 'Failed',
+                }
+            return response
+        elif run.status == "expired":
+            logger.warn(f'Run: {run.id} Thread: {assistant_thread.id}: expired...')
+            response = {
+                "role": "assistant",
+                "content": 'Expired',
+            }
+            return new_messages
```

Comment:
new_messages is not defined here.

Reply:
Fixed this in this PR: https://github.com/syscoin/autogen/pull/1 @sidhujag please take a look and add to your PR. I don't have write access to your repo.